### PR TITLE
Avoid galaxy install failure with already symlinked collections

### PR DESCRIPTION
**Pull request by ssbarnea**
Fixes: #165

### CODE REVIEW

File: src/ansible_compat/runtime.py [hunk 1]
```diff
@@ -352,6 +352,20 @@ def prepare_environment(
             return
 
         if os.path.exists("galaxy.yml"):
+            if destination:
```
Comment (zhan9san) [inline src/ansible_compat/runtime.py:355]: If the `destination` is None, the issue you want to work around also occur.

Do we need to consider the first default ansible_collection path(`~/.ansible/collections/`)?

File: src/ansible_compat/runtime.py [hunk 2]
```diff
@@ -352,6 +352,20 @@ def prepare_environment(
...
+                fqcn = fqcn_from_path(os.getcwd())
+                if (
+                    fqcn
+                    and os.path.islink("{destination}/{fqcn}")
```
Comment (zhan9san) [inline src/ansible_compat/runtime.py:359]: Do you miss `f`?

File: src/ansible_compat/loaders.py [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
     with open(filepath, encoding="utf-8") as content:
         return yaml.load(content, Loader=yaml.FullLoader)
+
+
+def fqcn_from_path(filepath: str) -> str | None:
+    """Return a FQCN from a path."""
+    galaxy_file = f"{filepath}/galaxy.yml"
+    if os.path.exists(galaxy_file):
+        galaxy = yaml_from_file(galaxy_file)
+        for k in ("namespace", "name"):
```
Comment (zhan9san) [inline src/ansible_compat/loaders.py:23]: Do we need to consider the `author` keyword?

It is used in `_get_galaxy_role_ns`
 https://github.com/ansible/ansible-compat/blob/aac5ae2bd68bcf09d1dd7c37066072df30d98c2c/src/ansible_compat/runtime.py#L585-L587

File: src/ansible_compat/runtime.py [hunk 4]
```diff
@@ -352,6 +352,20 @@ def prepare_environment(
...
+                    and os.path.realpath("{destination}/{fqcn}") != os.getcwd()
+                ):
+                    _logger.warning(
+                        "Collection is symlinked, but not pointing to %s directory. "
+                        "We will remove it.",
+                        os.getcwd(),
+                    )
+                    os.unlink("{destination}/{fqcn}")
```
Comment (zhan9san) [inline src/ansible_compat/runtime.py:367]: Do we need to raise an error here instead of deleting it?

File: src/ansible_compat/runtime.py (hunk 1 above)
Comment (ssbarnea) [reply src/ansible_compat/runtime.py:355]: Sorted the none. I am not sure yet about the default. I preferred to enable this logic only for the case when the destination is specified, to limit the risks.

File: src/ansible_compat/runtime.py (hunk 4 above)
Comment (ssbarnea) [reply src/ansible_compat/runtime.py]: Removing a symlink does not destroy data. We log the action.

File: src/ansible_compat/loaders.py (hunk 3 above)
Comment (ssbarnea) [reply src/ansible_compat/loaders.py:23]: No, our code here is only for collections. Author fallback is used on roles.

**Pull request merged**